DXRFixes/DeusEx/Classes/Carcass.uc

```diff
@@ -257,11 +257,22 @@ function Frob(Actor Frobber, Inventory frobWith)
 function bool TryLootItem(DeusExPlayer player, Inventory item)
 {
     local DeusExPickup invItem;
-    local int itemCount, newAmmoAmmout, ammoAddedAmount;
+    local int itemCount, newAmmoAmmout, ammoAddedAmount, action;
     local Weapon weap;
     local Ammo playerAmmo;
 
-    if (class'DXRLoadouts'.static.IsRefused(item.class)) {
+    action = class'DXRLoadouts'.static.GetLootAction(item.class);
+
+    if (
+        (action == 2 && DeusExPickup(item) != None) &&
+        (player.health != player.default.health || (player.energy != player.default.energy && HealingItem(item) != None))
```

Comment:
maybe a new function could make this easier, and then it could actually check the `energy` of the HealingItem, as well as the `balance` flag like HealingItem::DoHeal does

```suggestion
        (player.health < player.default.health || (HealingItem(item) != None && HealingItem(item).WouldHelp()))
```

Reply:
Would it be a problem to make food and soda `HealingItem`s? Then all this logic can live there.

Reply:
why not just make a function inside Carcass for `ItemWouldHelp` or something, it could also call a function in HealingItem, but do the simple checks for other items

Reply:
Is `Carcass` really the right place for a function like that? It doesn't necessarily have anything to do with a carcass.

Reply:
then use a static function